### PR TITLE
Add presenter code to media-service

### PR DESCRIPTION
**Pull request by reiterl**
Add presenter check code to auth and some cleanup.

### CODE REVIEW

File: src/auth.py [hunk 1]
```diff
@@ -25,13 +32,17 @@ def get_mediafile_id(meeting_id, path, app, cookie):
...
 
 
-def get_check_request_url():
-    presenter_host = "todo"
-    presenter_port = "todo"
+def get_presenter_url(meeting_id, path):
+    presenter_host = os.environ.get("PRESENTER_HOST")
+    presenter_port = os.environ.get("PRESENTER_PORT")
+    if presenter_host is None:
+        raise ServerError("PRESENTER_HOST is not set")
```
Comment (FinnStutzenstein) [inline src/auth.py:45]: Can you make this check early on startup? If the service is misconfigured, it should fail (exitcode >0) early and not on the first request. Maybe create a module, that handles all configuration variables in one place. The same should be done for the postgres configuration.

File: src/mediaserver.py [hunk 2]
```diff
@@ -48,6 +48,14 @@ def chunked(size, source):
     return Response(chunked(block_size, data), mimetype=mimetype)
 
 
+# for testing
```
Comment (FinnStutzenstein) [inline src/mediaserver.py:51]: THis has to be solved differently: First, do not include test code in the main code. Second, you have to test failure cases. What happens if it does not respond, respond wir errors or with invalid data.

Please find a way to mock the "auth-service" (auth.py) in your tests. maybe dependency injection helps.

File: src/auth.py (hunk 1 above)
Comment (reiterl) [reply src/auth.py]: Add the env options "PRESENTER_HOST" and "PRESENTER_PORT" to the config options and check them at start, also fix the config check mechanism.

File: src/mediaserver.py (hunk 2 above)
Comment (reiterl) [reply src/mediaserver.py:51]: How would dependency injection help here?

Comment (FinnStutzenstein) [reply src/mediaserver.py:51]: You can swap implementations at runtime

File: Dockerfile.dummy_presenter [hunk 3]
```diff
@@ -0,0 +1,12 @@
+FROM python:3.8.1
```
Comment (FinnStutzenstein) [inline Dockerfile.dummy_presenter:1]: Cool, can you move this Dockerfile into the dummy_presenter folder?

File: src/auth.py [hunk 4]
```diff
@@ -4,14 +4,19 @@
...
+        {
+            "presenter": "get_mediafile_id",
+            "data": {"meeting_id": meeting_id, "path": path},
+        }
+    ]
 
     try:
-        response = requests.post(check_request_url, headers={"Cookie": cookie})
+        response = requests.post(
+            presenter_url, headers={"Cookie": cookie}, json=payload
```
Comment (FinnStutzenstein) [inline src/auth.py:18]: Please pass all headers you get to the presenter service (except content type, content length, ...). There will be extra headers the client will send, that are needed for authentication. Some headers will be set by the HaProxy, that are also needed. Just forwarding them would be sufficient.

File: Dockerfile.dummy_presenter (hunk 3 above)
Comment (reiterl) [reply Dockerfile.dummy_presenter]: Done.

File: src/auth.py (hunk 4 above)
Comment (reiterl) [reply src/auth.py]: pass all headers to the presenter, except headers which contain content.

**Comment by FinnStutzenstein** [comment]
Ok, I tested the changes and they look good! One thing left: Test the response content in all tests. On errors, there should be a json response with a message what is wrong.

**Comment by reiterl** [comment]
Tests updated: include content check.
In error case a json with message is returned

**Pull request merged**